semantic_model_generator/sqlgen/generate_sql.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional, Set, Union
+
+import sqlglot
+from sqlglot.dialects.snowflake import Snowflake
+
+from semantic_model_generator.protos.semantic_model_pb2 import (
+    Dimension,
+    FullyQualifiedTable,
+    Measure,
+    Table,
+    TimeDimension,
+)
+
+
+def _fully_qualified_table_name(table: FullyQualifiedTable) -> str:
+    """Returns fully qualified table name such as my_db.my_schema.my_table"""
+    fqn = table.table
+    if len(table.schema) > 0:
+        fqn = f"{table.schema}.{fqn}"
+    if len(table.database) > 0:
+        fqn = f"{table.database}.{fqn}"
+    return fqn
+
+
+def _convert_to_snowflake_sql(sql: str) -> str:
+    """
+    Converts a given SQL statement to Snowflake SQL syntax using SQLGlot.
+
+    Args:
+    sql (str): The SQL statement to convert.
+
+    Returns:
+    str: The SQL statement in Snowflake syntax.
+    """
+    try:
+        expression = sqlglot.parse_one(sql, dialect=Snowflake)
+    except Exception as e:
+        raise ValueError(
+            f"Unable to parse sql statement.\n Provided sql: {sql}\n. Error: {e}"
+        )
+
+    return expression.sql()
+
+
+def _create_select_statement(table: Table, cols: Set[str], limit: int) -> str:
+    def _return_col_or_expr(
+        col: Union[TimeDimension, Dimension, Measure], cols: Set[str]
+    ) -> Optional[str]:
+        # TODO(jhilgart): Handle quoted names properly.
+        if col.name.lower() not in cols:
```

Comment:
I think we can drop this check and all the `cols: Str[str]` arguments for these methods.  Because here we just need to generate a select statement with one expression per logical column.

Reply:
makes sense